src/schema/nodes/table.ts

```diff
@@ -14,72 +14,218 @@
  * limitations under the License.
```

Comment:
Is this a copy from the prosemirror-tables?

Reply:
No, its not. 

I tried to use prosemirror's schema directly, but we use some custom attributes. The first solution was just to copy the schema and add our changes to it (similar to what we had going on before), but maintaining that will become difficult. 

The second solution was to add our custom attributes directly to the objects from the schema, for example replace the toDom() in the table schema to ours. this seemed reasonable but again, we'll have some issues of maintainability. Specially in table_cell & table_header, in toDOM(), we check if the node has textContent `node.textContent` and append  `placeholder` class if it doesn't. And in their implementation and what they expose, we can't check if the node has text content. The only control we have is in the tableOptions we pass to their library to generate the nodes. 

The third solution is what's currently implemented in this PR, it tries to maintain everything their schema has, even if it changes. And adds our own custom changes at the cost of over engineering and a lot of code. 

This is probably easier to explain in a call, so let me know if you have the time. Also please let me know if you prefer something else.